app/src/main/java/unipd/se18/ocrcamera/PreProcessing.java

```diff
@@ -0,0 +1,369 @@
+    }
+
+
+    /**
+     * Change the brightness of the image into an optimal one
+     * @param image the image we want to modify the brightness
+     * @return the image with the modified brightness
+     * @author Thomas Porro(g1), Giovanni Fasan(g1), Oscar Garrido(g1)
+     */
+    private Bitmap editBright(Bitmap image){
+        //Converts the image into a matrix
+        Mat bright;
+        try{
+            bright = IPUtils.conversionBitmapToMat(image);
+        } catch (ConversionFailedException error){
+            Log.e(TAG, error.getErrorMessage());
+            return image;
+        }
+
+        //Call the internal method isBright to detect if the image is bright or dark
+        //and change the brightness according to the number obtained
+        Mat modifiedMat = new Mat();
+        switch (isBright(bright)) {
+            case 1: //If the image is too bright
+                Log.d(TAG, "Case==1 ==> Too bright");
+                //Darkens the colour's brightness until it's in an optimal value
+                for(double changeBrightness=0; changeBrightness!=-240; changeBrightness-=15){
+                    //Converts an array to another data type with optional scaling.
+                    bright.convertTo(modifiedMat, -1, 1, changeBrightness);
+                    if(isBright(modifiedMat)==0){
+                        try{
+                            return IPUtils.conversionMatToBitmap(modifiedMat);
+                        } catch (ConversionFailedException error){
+                            Log.e(TAG, error.getErrorMessage());
+                            return image;
+                        }
+                    }
+                }
+                break;
+
+            case 2: //If the image is too dark
+                Log.d(TAG, "Case==2 ==> Too dark");
+                //Lightens the colour's brightness until it's in an optimal value
+                for(double changeBrightness=0; changeBrightness!=240; changeBrightness+=15){
+                    //Converts an array to another data type with optional scaling.
+                    bright.convertTo(modifiedMat, -1, 1, changeBrightness);
+                    if(isBright(modifiedMat)==0){
+                        //If the conversion failed it returns the original image
+                        try{
+                            return IPUtils.conversionMatToBitmap(modifiedMat);
+                        } catch (ConversionFailedException error){
+                            Log.e(TAG, error.getErrorMessage());
+                            return image;
+                        }
+                    }
+                }
+                break;
+
+            case 0: //Image is neither too bright nor too dark
+                Log.d(TAG, "Case==0 ==> Perfect image");
+                return image;
+        }
+        return image;
+    }
+
+
+    /**
+     * @author Thomas Porro(g1), Oscar Garrido (g1), Giovanni Fasan(g1).
+     * See PreProcessingMethods.java
```

Comment:
I appreciate you for this indication, in general I think that indication for coder or reviewer clarify the code.
If you would like to use the javadoc method to do this, in this case, it's similar to this {@link PreProcessingMethods}